### ISSUE TITLE
At first download, ask to choose beetween "internal" and "external"

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#2068]
This is an idea to avoid users to download to the "wrong" place because for the beginners it is not obvious what are all the options the app provides.

**Comment by ARJUPTA** [comment]
I think it's a decent feature request. Can i work on it?

**Comment by kelson42** [comment]
@s-ayush2903 You volunteer to fix it ? ;)

**Comment by s-ayush2903** [comment]
@kelson42 Well, I believe that people who have contributing for some long time to the project should not take up/fix the `good-first-issues` as I don't want to real beginners to struggle with our project. If I or anyone(who has been contributing to the project for some time) take up and finish all the low hanging fruits then it would become nearly impossible for beginners in on-boarding to our project. We'll definitely get new contributors as time progresses towards GSoC and I'm waiting for those beginners to take these up. I don't want to snatch opportunity from 'em.

**Comment by isha-24** [comment]
@s-ayush2903 I would like to work on this issue. Could you pls tell me how can I proceed with it?

**Comment by s-ayush2903** [comment]
@isha-24 You should create a new shared preference for first-time download(could be with name: _isFirstDownload_) and set it to true after the download is finished for the very first time.

**Comment by gouri-panda** [comment]
@isha-24 go ahead! You are assigned.

**Comment by kelson42** [comment]
@isha-24 Any news about your effort/patch?

**Comment by markZuckerberg2** [comment]
Can I work on this?

**Comment by kelson42** [comment]
@markZuckerberg2 Yes, but before starting codeing, please explain here how your solution will look like.

**Comment by yyprince334** [comment]
I would like to contribute in this issue.

**Comment by yyprince334** [comment]
Would you suggest me , how to start

**Comment by gouri-panda** [comment]
In `OnLibraryFragment` When we select a book item i.e `onBookItemClick()`, If the user downloading a zim file for the first time ask them where to put the zim file(i.e in internal or external). We already have a method i.e `showStorageSelectDialog`. It  can be done by `SharedPreference`

**Comment by gouri-panda** [comment]
@kelson42 I'm not sure which one to give this task. Can you help me here, please ;)

**Comment by kelson42** [comment]
@gouri-panda Usually, this is first come, first served ;)
@ yyprince334 @markZuckerberg2 You are welcome, but would you be able please first to explain in a few sentences your approach to solve the problem?

**Comment by markZuckerberg2** [comment]
@kelson42 Can I take the above approach?

**Comment by Rishavgupta12345** [comment]
is this issue still pending ? 

can I work on it ?

**Comment by gouri-panda** [comment]
@Rishavgupta12345 go ahead!

**Comment by cdhiraj40** [comment]
@kelson42 if this issue is still relevant and to be worked on I would like to work on it :)

**Comment by kelson42** [comment]
@cdhiraj40 Yes, this issue is still pertinent. Thx for proposing yourself.

**Comment by cdhiraj40** [comment]
hey @kelson42, do assign me this issue  :)

**Comment by the-illuminatus** [comment]
if this issue is still relevant and to be worked upon, I would like to contribute my bit on it.

**Comment by cdhiraj40** [comment]
hey @the-illuminatus the pr has already been made for this issue. I will suggest you pick up another issue, if any help needed then do tell :)

**Comment by the-illuminatus** [comment]
> hey @the-illuminatus the pr has already been made for this issue. I will suggest you pick up another issue, if any help needed then do tell :)

yes please can you suggest me a issue wherein i can contribute since most of the issues have been closed and i am not able to get started with the PRs plz suggest me a issue based on good first issue.

**Comment by cdhiraj40** [comment]
> > hey @the-illuminatus the pr has already been made for this issue. I will suggest you pick up another issue, if any help needed then do tell :)
> 
> yes please can you suggest me a issue wherein i can contribute since most of the issues have been closed and i am not able to get started with the PRs plz suggest me a issue based on good first issue.

sure!
